lib_async/kafka_async.mli

```diff
@@ -0,0 +1,28 @@
+open Async
+
+type producer
+
+type consumer
+
+type 'a response = ('a, Kafka.error * string) result
+
+val produce :
+  producer ->
```

Comment:
I would expect something more consistent with the Lwt and Sync interfaces. These two don't need a producer. Idem for `rd_kafka_produce`. Why is this required for Async?

Reply:
I don't particularly like the idea of having mutable state on the module level (where `pending_msg` is in the Lwt bindings) so I opted to move it to the `producer` type, so multiple producers are independent and GCing a producer also cleans out the table.

Reply:
Good points. I will have a look to see what can be done with the Lwt and Sync interfaces.